api/api.yaml

```diff
@@ -0,0 +1,220 @@
+openapi: 3.1.0
+info:
+  title: Typeberry Codec API
+  version: 1.0.0
+servers:
+  - url: /api/v1
+paths:
+  /types:
+    get:
+      summary: "An overview of all available types. Only a subset of types is exposed here for testing. More will follow."
+      operationId: getTypes
+      responses:
+        "200":
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Type"
+      tags:
+      - Types
+  /types/{id}/codec_to_json:
+    post:
+      description: "This decodes the CODEC value into the given type and then re-encodes it as JSON."
+      operationId: codecToJson
+      parameters:
+      - description: Type name
+        example: Header
+        in: path
+        name: id
+        required: true
+        schema:
+          enum: ['Header', 'Block', 'Extrinsic', 'EpochMarker', 'AvailabilityAssurance', 'AssurancesExtrinsic', 'Culprit', 'Fault', 'Judgement', 'Verdict', 'DisputesExtrinsic', 'Credential', 'ReportGuarantee', 'GuaranteesExtrinsic', 'Preimage', 'PreimageExtrinsic', 'RefineContext', 'SignedTicket', 'Ticket', 'TicketExtrinsic', 'ImportSpec', 'WorkItem', 'WorkItemExtrinsicSpec', 'WorkPackage', 'WorkPackageSpec', 'WorkReport', 'WorkExecResult', 'WorkResult']
```

Comment:
Is there a way we can verify this is matching what we have in `api/src/kind.rs`?

Reply:
fixed

Reply:
added validation